python/plugins/processing/algs/qgis/FieldPyculator.py

```diff
@@ -59,16 +61,31 @@ def __init__(self):
         super().__init__()
 
     def initAlgorithm(self, config=None):
-        self.type_names = [self.tr('Integer'),
-                           self.tr('Float'),
-                           self.tr('String')]
-
         self.addParameter(QgsProcessingParameterFeatureSource(self.INPUT, self.tr('Input layer'),
                                                               types=[QgsProcessing.TypeVector]))
         self.addParameter(QgsProcessingParameterString(self.FIELD_NAME,
                                                        self.tr('Result field name'), defaultValue='NewField'))
-        self.addParameter(QgsProcessingParameterEnum(self.FIELD_TYPE,
-                                                     self.tr('Field type'), options=self.type_names))
+
+        types = [[QVariant.Int, QVariant.Invalid],
+                 [QVariant.Double, QVariant.Invalid],
+                 [QVariant.String, QVariant.Invalid],
+                 [QVariant.Bool, QVariant.Invalid],
+                 [QVariant.Date, QVariant.Invalid],
+                 [QVariant.Time, QVariant.Invalid],
+                 [QVariant.DateTime, QVariant.Invalid],
+                 [QVariant.ByteArray, QVariant.Invalid],
+                 [QVariant.StringList, QVariant.String],
+                 [QVariant.List, QVariant.Int],
+                 [QVariant.List, QVariant.Double]]
```

Comment:
```suggestion
        types = [(QVariant.Int, QVariant.Invalid),
                 (QVariant.Double, QVariant.Invalid),
                 (QVariant.String, QVariant.Invalid),
                 (QVariant.Bool, QVariant.Invalid),
                 (QVariant.Date, QVariant.Invalid),
                 (QVariant.Time, QVariant.Invalid),
                 (QVariant.DateTime, QVariant.Invalid),
                 (QVariant.ByteArray, QVariant.Invalid),
                 (QVariant.StringList, QVariant.String),
                 (QVariant.List, QVariant.Int),
                 (QVariant.List, QVariant.Double)]
```

Good case for tuples